### PR TITLE
Unify config

### PR DESCRIPTION
**Pull request by henryyang42**
Move mysql settings and email into config.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,10 +32,7 @@ git clone git@github.com:bruce3557/NTHUOJ_web.git
     python install.py
     ```
 
-* After installing, you can modify `'<project/root>/nthuoj.ini'` if your database setting is modified.
-
-
-* After installing, you can modify `'<project/root>/emailInfo.py'` if your email host setting is modified.
+* After installing, you can modify `'<project/root>/config/nthuoj.cfg'` if your want to change project configurations.
```
Comment (nickelcheng) [inline README.md]: I think the path is ```project_root/nthuoj/config/nthuoj.cfg```

File: install.py [hunk 2]
```diff
@@ -21,37 +21,51 @@
...
-    write_email_file(email_host, email_host_pwd)
+    email_host_pwd = getpass.getpass("Email host's password : ")
+    write_email_config(config, email_host, email_host_pwd)
+    print '========================================'
 
-# Database Migratinos
-db_migrate()
-
-# Create super user
-ans = raw_input('Create super user?[Y/n] ')
```
Comment (nickelcheng) [inline install.py]: Why don't you create a super user in installer?

File: func.py [hunk 3]
```diff
@@ -24,26 +24,32 @@
...
+        config.set('path', key, paths[key])
+
 
 def django_manage(args):
     cmd = 'python ./manage.py ' + args
     os.system(cmd)
 
+
 def db_migrate():
     apps = ['index', 'problem', 'users', 'contest', 'team', 'group']
```
Comment (nickelcheng) [inline func.py]: I think we can remove migration folders in this function. (It may avoid some trouble)

File: install.py (hunk 2 above)
Comment (henryyang42) [inline install.py]: My bad.

File: install.py [hunk 4]
```diff
@@ -21,37 +21,57 @@
...
-    email_host_pwd = getpass.getpass('Email host\'s password : ')
-    write_email_file(email_host, email_host_pwd)
+    email_host_pwd = getpass.getpass("Email host's password : ")
+    write_email_config(config, email_host, email_host_pwd)
+    print '========================================'
 
-# Database Migratinos
-db_migrate()
+# Change defaut path
+ans = raw_input('Customize source code, testcase path? [Y/n] ')
```
Comment (nickelcheng) [inline install.py]: Maybe we can tell user our default value

Comment (henryyang42) [inline install.py]: Added.

**Comment by nickelcheng** [comment]
Please install bower in ```install.py```
+1 beside the comment. @bruce3557 please check.

File: nthuoj/settings.py [hunk 5]
```diff
@@ -118,7 +117,9 @@
 
 EMAIL_USE_TLS = True
 EMAIL_HOST = 'smtp.gmail.com'
-EMAIL_PORT = 587
+EMAIL_HOST_USER = get_config('email', 'user')
+EMAIL_HOST_PASSWORD = get_config('email', 'password')
+EMAIL_PORT = 689
```
Comment (bruce3557) [inline nthuoj/settings.py]: XD

**Comment by bruce3557** [comment]
+1, going to merge

File: nthuoj/settings.py (hunk 5 above)
Comment (henryyang42) [inline nthuoj/settings.py]: It turned out that 689 is useless QQ.

**Pull request merged**